base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -2935,6 +2971,81 @@ where
         self.import_onesided_outputs(scanned_outputs).await
     }
 
+    // Scanning outputs addressed to this wallet
+    #[allow(clippy::too_many_lines)]
+    async fn scan_outputs_for_multisig(
+        &mut self,
+        outputs: Vec<(TransactionOutput, Option<TxId>)>,
+    ) -> Result<Vec<RecoveredOutput>, OutputManagerError> {
+        // 1. Get all your wallet's public keys (or just the spend key for now)
+        let mut scanned_outputs = vec![];
+
+        for (output, tx_id) in outputs {
+            // 2. Check if the script is a multisig script
+            if let Some(Opcode::CheckMultiSigVerify(_m, _n, pubkeys, _msg)) = output
```

Comment:
this will catch outputs that perhaps dont match it, 
scan for very spesific ones, 
in this case, the scripts should be 
opcode[0] = CheckMultiSigVerify
opcode[1] = pushpubkey
and check them all to ensure they are correct